finance_manager/src/main/java/com/manager/finance/wallet/dao/WalletDao.java

```diff
@@ -5,14 +5,16 @@
 import com.manager.finance.wallet.model.entity.api.DefaultCurrency;
 import java.util.List;
 import java.util.Optional;
+import org.springframework.data.domain.Sort;
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface WalletDao extends JpaRepository<Wallet, Long> {
-    List<Wallet> findWalletByProfileId(Profile profileId);
+    List<Wallet> findWalletByProfileIdAndIsDeleted(Profile profileId, Boolean isDeleted, Sort sort);
```

Comment:
if, in general, we want to select only active wallets, I would suggest using @Where  annotation on entity

@Where(clause = "is_deleted=false")